packages/babel-generator/src/node/parentheses.js

```diff
@@ -166,20 +166,22 @@ export function UnaryLike(node: Object, parent: Object): boolean {
 }
 
 export function FunctionExpression(node: Object, parent: Object, printStack: Array<Object>): boolean {
-  return isFirstInStatement(printStack, { considerDefaultExports: true });
-}
-
-export function ArrowFunctionExpression(node: Object, parent: Object): boolean {
-  // export default (function () {});
-  if (t.isExportDeclaration(parent)) {
+  if (t.isTaggedTemplateExpression(parent)) {
```

Comment:
I think the function expression case is fine, it's only the arrow function that needs fixing.